File: ios/Classes/Transfer/TrnDirectHandler.m
#import "TrnDirectHandler.h"


@implementation TrnDirectHandler

- (void)start:(UIViewController *)controller dictionary:(NSDictionary *) dictionary delegate:(id<P24TransferDelegate>) delegate {
      bool isSandbox = [[dictionary objectForKey:@"isSandbox"] boolValue];
      NSDictionary* transactionParamsDictionary = [dictionary objectForKey:@"transactionParams"];
      
      P24TransactionParams * transactionParams = [TransactionParamsMapper buildTransaction: transactionParamsDictionary];
      P24TrnDirectParams* params = [[P24TrnDirectParams alloc] initWithTransactionParams: transactionParams];
      params.sandbox = isSandbox;
      
      [P24 startTrnDirect:params inViewController: controller delegate: delegate];
}

@end

File: ios/Classes/Transfer/TrnDirectHandler.swift
import P24

internal struct TrnDirectHandler : TransferHandler {
    internal func start(controller: UIViewController?, dictionary: [String: Any], delegate: P24TransferDelegate) {
        let transactionParams = TransactionParamsMapper.buildTransaction(dictionary["transactionParams"] as! [String: Any])
        let params = P24TrnDirectParams(transactionParams: transactionParams)!
        params.sandbox =  dictionary["isSandbox"] as! Bool
        P24.startTrnDirect(params, in: controller, delegate: delegate)
    }
}

